### ISSUE TITLE
Application not responding when load the long content

### ISSUE THREAD

**Issue by adrian09h** [noties/Markwon#229]
* **Markwon version**: 3

I am getting the application not responding for about 4 seconds when load the this content.
I am using the recycler view as markdown view.
How can I load this content smoothly?

$$ P_1 + \frac{1}{2}mv_1^2 + pgh_1 = P_2 + \frac{1}{2}mv_2^2 + pgh_2 +  P_1 + \frac{1}{2}mv_1^2 + pgh_1 = P_2 + \frac{1}{2}mv_2^2 + pgh_2 +  P_1 + \frac{1}{2}mv_1^2 + pgh_1 = P_2 + \frac{1}{2}mv_2^2 + pgh_2 +  P_1 + \frac{1}{2}mv_1^2 + pgh_1 = P_2 + \frac{1}{2}mv_2^2 + pgh_2$$

# Upmath: Math Online Editor
### _Create web articles and&nbsp;blog posts with&nbsp;equations and&nbsp;diagrams_

Upmath extremely simplifies this task by using Markdown and LaTeX. It converts the Markdown syntax extended with LaTeX equations support into HTML code you can publish anywhere on the web.

![Paper written in LaTeX](https://upload.wikimedia.org/wikipedia/commons/thumb/2/28/Salma_Hayek_by_Gage_Skidmore_2.jpg/640px-Salma_Hayek_by_Gage_Skidmore_2.jpg)

## Markdown

Definition from [Wikipedia](https://en.wikipedia.org/wiki/Markdown):

> Markdown is a lightweight markup language with plain text formatting syntax designed so that it can be converted to HTML and many other formats using a tool by the same name. Markdown is often used to format readme files, for writing messages in online discussion forums, and to create rich text using a plain text editor.

The main idea of Markdown is to use a simple plain text markup. It's ~~hard~~ easy to __make__ **bold** _or_ *italic* text. Simple equations can be formatted with subscripts and superscripts: *E*~0~=*mc*^2^. I have added the LaTeX support: $$E_0=mc^2$$.

Among Markdown features are:

* images (see above);
* links: [service main page](/ "link title");
* code: `untouched equation source is *E*~0~=*mc*^2^`;
* unordered lists--when a line starts with `+`, `-`, or `*`;
  1. sub-lists
  1. and ordered lists too;
* direct use <nobr>of HTML</nobr>&ndash;for <span style="color: red">anything else</span>. 

Also the editor supports typographic replacements: (c) (r) (tm) (p) +- !!!!!! ???? ,,  -- ---

## LaTeX

The editor converts LaTeX equations in double-dollars `$$`: $$ax^2+bx+c=0$$. All equations are rendered as block equations. If you need inline ones, you can add the prefix `\inline`: $$p={1\over q}$$. But it is a good practice to place big equations on separate lines:

$$x_{1,2} = {-b\pm\sqrt{b^2 - 4ac} \over 2a}.$$

In this case the LaTeX syntax will be highlighted in the source code. You can even add equation numbers (unfortunately there is no automatic numbering and refs support):

$$|\vec{A}|=\sqrt{A_x^2 + A_y^2 + A_z^2}.$$(1)

It is possible to write Cyrillic symbols in `\text` command: $$Q_\text{плавления}>0$$.

One can use matrices:

$$T^{\mu\nu}=\begin{pmatrix}
\varepsilon&0&0&0\\
0&\varepsilon/3&0&0\\
0&0&\varepsilon/3&0\\
0&0&0&\varepsilon/3
\end{pmatrix},$$

integrals:

$$P_\omega={n_\omega\over 2}\hbar\omega\,{1+R\over 1-v^2}\int\limits_{-1}^{1}dx\,(x-v)|x-v|,$$

and [the rest of LaTeX features](https://en.wikibooks.org/wiki/LaTeX/Mathematics).

## About Upmath

It works in browsers, except equations rendered [on the server](//tex.s2cms.com/). The editor stores your text in the browser to prevent the loss of your work in case of software or hardware failures.

I have designed and developed this lightweight editor and the service for converting LaTeX equations into svg-pictures to make publishing math texts on the web easy. I consider client-side rendering, the rival technique implemented in [MathJax](https://www.mathjax.org/), to be too limited and resource-consuming, especially on mobile devices.

The source code is [published on Github](https://github.com/parpalak/upmath.me) under MIT license.

***

Now you can erase this instruction and start writing your own scientific post. If you want to see the instruction again, open the editor in a private tab, in a different browser or download and clear your post and refresh the page.

Have a nice day :)

[Roman Parpalak](https://written.ru/), web developer and UX expert.

The *Gamma function* satisfying $\Gamma(n) = (n-1)!\quad\forall n\in\mathbb N$ is via the Euler integral

$$
\Gamma(z) = \int_0^\infty t^{z-1}e^{-t}dt\,.
$$

**Comment by noties** [comment]
Hello @adrian09h !

I would suggest offloading your main thread and parse markdown in background. On the other hand markdown parsing is quite fast and your input doesn't seem like too complicated nor long. So there can be an error in your initialization code.

**Comment by adrian09h** [comment]
Hi Thank you for your reply.
I have one issue.
When I load the image using Glide plugin, The memory leak issue happens.
Whenever I am entering one activity with image, memory is increasing without releasing.
Why this is happened?

**Comment by noties** [comment]
Hello @adrian09h !

Are you sure that it is a memory leak? Did you check it with a `LeakCanary` or some other tool? Have you tried forcing garbage collection?

Glide, most likely, has some memory cache, which is released when needed and not the moment when resource you are using is detached from UI
